torch/ao/quantization/fx/prepare.py

```diff
@@ -1191,8 +1270,27 @@ def insert_observers_for_model(
                 not node.op == 'output'
             )
 
-            is_supported_by_backend = is_pattern_dtype_config_supported_by_backend(
-                pattern, matched_node_pattern, node_name_to_target_dtype, backend_config)
+            # Check whether this pattern's dtype is supported by the backend
+            if pattern is not None and matched_node_pattern is not None and backend_config is not None:
+                supported_dtype_config = _get_supported_pattern_dtype_config(
+                    pattern, matched_node_pattern, node_name_to_target_dtype, backend_config)
+                is_supported_by_backend = supported_dtype_config is not None
+            else:
+                supported_dtype_config = None
+                is_supported_by_backend = True
+
+            # Update the QConfig subject to constraints from the backend, if any
+            if qconfig is not None and supported_dtype_config is not None:
+                new_qconfig = _update_qconfig_based_on_dtype_config(qconfig, supported_dtype_config, pattern)
```

Comment:
should we be updating user's qconfigs?

Reply:
This is actually returning a new qconfig. We are not mutating the user's qconfigs (they are immutable anyway). Maybe the naming is confusing here, so I updated it to `_get_qconfig_compatible_with_dtype_config`

Reply:
should we do this implicitly or should we just error out and ask user to update their qconfigs?

Reply:
OK, let's move this conversation to Vasiliy's comment below. I think he raised a similar question

Reply:
Updated to just throw the error